crates/ol/state-types/src/writebatch.rs

```diff
@@ -0,0 +1,50 @@
+use std::collections::HashMap;
+
+use strata_acct_types::{AccountId, AccountSerial, BitcoinAmount, Mmr64};
+use strata_identifiers::{Buf32, EpochCommitment};
+use strata_ledger_types::AsmManifest;
+use strata_snark_acct_types::MessageEntry;
+
+use crate::AccountState;
+
+/// Batch of state changes from executing a block
+#[derive(Clone, Debug)]
+pub struct WriteBatch {
+    // Global state changes
+    pub new_slot: u64,
+
+    // L1 View / Epochal state changes
+    pub l1_view_writes: L1ViewWrites,
+
+    // Account changes - store full AccountState for now
+    // TODO: Consider storing diffs instead of full state for efficiency
+    pub new_accounts: Vec<(AccountId, AccountSerial, AccountState)>,
```

Comment:
* This should be structured better.
* Diffs are not relevant or needed here.
* Serials are redundant.